src/crimson/os/seastore/seastore_types.h

```diff
@@ -1321,15 +1322,23 @@ struct segment_tail_t {
 
   journal_seq_t journal_tail;
   segment_nonce_t segment_nonce;
+
+  segment_type_t type;
+
```

Comment:
Why need the type in `segment_tail_t`, duplicating the information in `segment_header_t` ?

Reply:
Looks some existing logic relies on `segment_tail_t` to get the `segment_type_t`. I think that might not be necessary. But that is not related to this PR.

Reply:
I just wanted to make sure the header and tail as consistent as possible.